contracts/contracts/UserReplicaSetManager.sol

```diff
@@ -0,0 +1,428 @@
+     * @param _newCnodeId - Incoming spID
+     * @param _newCnodeDelegateOwnerWallet - Incoming SP delegateOwnerWallet
+     * @param _proposerSpIds - Array of 3 spIDs proposing new node
+     * @param _proposerNonces - Array of 3 nonces, each index corresponding to _proposerSpIds
+     * @param _proposer1Sig - Signature from first proposing node
+     * @param _proposer2Sig - Signature from second proposing node
+     * @param _proposer3Sig - Signature from third proposing node
+     */
+    function addOrUpdateContentNode(
+        uint _newCnodeId,
+        address _newCnodeDelegateOwnerWallet,
+        uint[3] calldata _proposerSpIds,
+        bytes32[3] calldata _proposerNonces,
+        bytes calldata _proposer1Sig,
+        bytes calldata _proposer2Sig,
+        bytes calldata _proposer3Sig
+    ) external
+    {
+        _requireSeed();
+        // For every entry in spIds/Nonces/Sigs
+        //  Recover signer using the signature for inner function (tmp is addOrUpdateCreatorNode)
+        //  Confirm that the spId <-> recoveredSigner DOES exist and match what is stored on chain
+        address proposer1Address = _recoverProposeAddOrUpdateContentNodeSignerAddress(
+            _newCnodeId,
+            _newCnodeDelegateOwnerWallet,
+            _proposerSpIds[0],
+            _proposerNonces[0],
+            _proposer1Sig
+        );
+        address proposer2Address = _recoverProposeAddOrUpdateContentNodeSignerAddress(
+            _newCnodeId,
+            _newCnodeDelegateOwnerWallet,
+            _proposerSpIds[1],
+            _proposerNonces[1],
+            _proposer2Sig
+        );
+        address proposer3Address = _recoverProposeAddOrUpdateContentNodeSignerAddress(
+            _newCnodeId,
+            _newCnodeDelegateOwnerWallet,
+            _proposerSpIds[2],
+            _proposerNonces[2],
+            _proposer3Sig
+        );
+        _validateUpdateOperation(
+            proposer1Address,
+            proposer2Address,
+            proposer3Address,
+            _proposerSpIds
+        );
+        spIdToContentNodeDelegateWallet[_newCnodeId] = _newCnodeDelegateOwnerWallet;
+        emit AddOrUpdateContentNode(
+            _newCnodeId,
+            _newCnodeDelegateOwnerWallet,
+            _proposerSpIds,
+            proposer1Address,
+            proposer2Address,
+            proposer3Address
+        );
+    }
+
+    /**
+     * @notice Function used to perform updates to a given user's replica set
+     * @param _userId - User for whom this update operation is being performed
+     * @param _primary - Incoming primary for _userId
+     * @param _secondaries - Incoming array of n spIDs for _userId
+     * @param _oldPrimary - Current primary on chain for _userId
+     * @param _oldSecondaries - Current array of secondaries on chain for _userId
+     * @param _requestNonce - Nonce generated by signer
+     * @param _subjectSig - Signature generated by signer
+     */
+    function updateReplicaSet(
```

Comment:
some comments / documentation explaining who should be able to submit this request and why would be helpful